app/src/main/java/com/rajatgoyal/puzzle15/adapter/HighScoresAdapter.java

```diff
@@ -18,44 +22,59 @@
 
 public class HighScoresAdapter extends RecyclerView.Adapter<HighScoresAdapter.HighScoreViewHolder> {
 
-    private ArrayList<HighScore> highScores;
+    private ArrayList<GamePlay> gamePlays;
```

Comment:
You can instantiate this object, so at least NPE is out of play, what if you forget to call `setGamePlays` method